Lib/ufo2ft/featureWriters/kernFeatureWriter.py

```diff
@@ -99,13 +99,89 @@ class KernFeatureWriter(BaseFeatureWriter):
+      kernings that reference unknown glyphs.
+    * Determine which scripts the kerning affects (read: "the font most probably
+      supports"), to know which lookups to generate later:
+        * First, determine the unambiguous script associations for each
+          (Unicoded) glyph in the glyphset, as in, glyphs that have a single
+          entry for their Unicode script extensions property;
+        * then, parse the `languagesystem` statements in the provided feature
+          file to add on top.
+    * Compile a Unicode cmap from the UFO and a GSUB table from the features so
+      far, so we can determine:
+        * the script (extensions) for each glyph in the glyphset, including
+          glyphs reachable via substitution, using the fontTools subsetter with
+          its `closure_glyphs` machinery;
+        * and the bidirectionality class, so we can later filter out kerning
+          pairs that would mix RTL and LTR glyphs, which will not occur in
+          applications. Unicode BiDi classes L, AN and ER are considered L, R
+          and AL are considered R.
+    * Get the kerning groups from the UFO and filter out glyphs not in the
+      glyphset and empty groups. Remember which group a glyph is a member of,
+      for kern1 and kern2, so we can later reconstruct per-script groups.
+    * Get the bare kerning pairs from the UFO, filtering out pairs with unknown
+      groups or glyphs not in the glyphset and (redundant) zero class-to-class
+      kernings and optionally quantizing kerning values.
+    * Start generating lookups. By default, the ignore marks flag is added to
+      each lookup. Kerning pairs that kern bases against marks or marks against
+      marks, according to the glyphs' GDEF category, then get split off into a
+      second lookup without the ignore marks flag.
+    * Go through all kerning pairs and split them up by script, to put them in
+      different lookups. This reduces the size of each lookup compared to
+      splitting by direction, as previously done.
+        * Partition the first and second side of a pair by script and emit only
+          those with the same script (e.g. `a` and `b` are both "Latn", `period`
+          and `period` are both "Common", but `a` and `a-cy` would mix "Latn"
+          and "Cyrl" and are dropped) or those that kern an explicit against a
+          "common" or implicit script, e.g. `a` and `period`.
+        * Glyphs can have multiple scripts assigned to them (legitimately, e.g.
+          U+0951 DEVANAGARI STRESS SIGN UDATTA, or for random reasons like
+          having both `sub h by h.sc` and `sub Etaprosgegrammeni by h.sc;`).
+          Only scripts that were determined earlier to be supported by the font
+          will be considered. Usually, we will emit pairs where both sides have
+          the same script and no splitting is necessary. The only mixed script
+          pairs we emit are common (e.g. Zyyy) against explicit (e.g. Latn)
```

Comment:
> The only mixed script pairs we emit are common (e.g. Zyyy) against explicit (e.g. Latn)

not only Zyyy (Common) and also Zinh (Inherited), no?

Reply:
Only Zyyy I think. I'm unsure about Zinh. Need to make a test case maybe...